src/System.Private.CoreLib/shared/System/TimeSpan.cs

```diff
@@ -288,13 +289,14 @@ public static TimeSpan FromTicks(long value)
             return new TimeSpan(value);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static long TimeToTicks(int hour, int minute, int second)
         {
             // totalSeconds is bounded by 2^31 * 2^12 + 2^31 * 2^8 + 2^31,
             // which is less than 2^44, meaning we won't overflow totalSeconds.
             long totalSeconds = (long)hour * 3600 + (long)minute * 60 + (long)second;
             if (totalSeconds > MaxSeconds || totalSeconds < MinSeconds)
```

Comment:
```suggestion
             if ((ulong)(totalSeconds - MinSeconds) > (MaxSeconds - MinSeconds))
```

Reply:
thanks for the suggestion but I think that it would rather not increase the perf but decrease the readability of the source code